src/Block/Types.hs

```diff
@@ -62,23 +89,26 @@ instance (ToJSON args, ToJSON mirror) => ToJSON (OgmiosRequest args mirror) wher
   toJSON =
     Aeson.genericToJSON Aeson.defaultOptions {Aeson.fieldLabelModifier = drop 1}
 
-type OgmiosFindIntersectRequest = OgmiosRequest CursorPoints OgmiosMirror
+type OgmiosFindIntersectRequest = OgmiosRequest Text OgmiosMirror
 
 type OgmiosRequestNextRequest = OgmiosRequest (Map Text Text) OgmiosMirror
 
 mkFindIntersectRequest :: BlockInfo -> OgmiosFindIntersectRequest
-mkFindIntersectRequest (BlockInfo firstBlockSlot firstBlockId) =
+mkFindIntersectRequest blockInfo =
   OgmiosRequest
     { _type = "jsonwsp/request"
     , _version = "1.0"
     , _servicename = "ogmios"
     , _methodname = "FindIntersect"
-    , _args = points
+    , _args = payload
     , _mirror = 0
     }
   where
-    points =
-      CursorPoints [CursorPoint (fromIntegral firstBlockSlot) firstBlockId]
+    points s i = CursorPoints [CursorPoint (fromIntegral s) i]
+    payload = decodeUtf8 $
+      toStrict $ case blockInfo of
+        BlockInfo slot hash -> Aeson.encode $ points slot hash
+        BlockOrigin -> Aeson.encode [BlockOrigin]
```

Comment:
Maybe it will be better to extract this logic from the function to the standalone `ToJSON a` implementation?

Reply:
The points encoding of the non-origin payload doesn't correspond to its json representation.